runtime/src/sign_with_exported_ecdsa.rs

```diff
@@ -0,0 +1,116 @@
+// Licensed under the Apache-2.0 license
+
+use crate::{dpe_crypto::DpeCrypto, Drivers};
+
+use caliptra_cfi_derive_git::cfi_impl_fn;
+use caliptra_cfi_lib_git::{cfi_assert, cfi_assert_eq, cfi_launder};
+
+use caliptra_common::mailbox_api::{
+    MailboxResp, MailboxRespHeader, SignWithExportedEcdsaReq, SignWithExportedEcdsaResp,
+};
+use caliptra_error::{CaliptraError, CaliptraResult};
+
+use crypto::{Crypto, Digest, EcdsaPub, EcdsaSig};
+use dpe::{DPE_PROFILE, MAX_EXPORTED_CDI_SIZE};
+use zerocopy::{FromBytes, IntoBytes};
+
+pub struct SignWithExportedEcdsaCmd;
+impl SignWithExportedEcdsaCmd {
+    /// SignWithExported signs a `digest` using an ECDSA keypair derived from a exported_cdi
+    /// handle and the CDI stored in DPE.
+    ///
+    /// # Arguments
+    ///
+    /// * `env` - DPE environment containing Crypto and Platform implementations
+    /// * `digest` - The data to be signed
+    /// * `exported_cdi_handle` - A handle from DPE that is exchanged for a CDI.
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    fn ecdsa_sign(
+        env: &mut DpeCrypto,
+        digest: &Digest,
+        exported_cdi_handle: &[u8; MAX_EXPORTED_CDI_SIZE],
+    ) -> CaliptraResult<(EcdsaSig, EcdsaPub)> {
+        let algs = DPE_PROFILE.alg_len();
+        let key_pair = env.derive_key_pair_exported(
+            algs,
+            exported_cdi_handle,
+            b"Exported ECC",
+            b"Exported ECC",
+        );
+
+        if cfi_launder(key_pair.is_ok()) {
+            #[cfg(not(feature = "no-cfi"))]
+            cfi_assert!(key_pair.is_ok());
+        } else {
+            #[cfg(not(feature = "no-cfi"))]
+            cfi_assert!(key_pair.is_err());
+        }
+        let (priv_key, pub_key) = key_pair
+            .map_err(|_| CaliptraError::RUNTIME_SIGN_WITH_EXPORTED_ECDSA_KEY_DERIVIATION_FAILED)?;
+
+        let sig = env
+            .ecdsa_sign_with_derived(algs, digest, &priv_key, &pub_key)
+            .map_err(|_| CaliptraError::RUNTIME_SIGN_WITH_EXPORTED_ECDSA_SIGNATURE_FAILED)?;
+
+        Ok((sig, pub_key))
+    }
+
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    #[inline(never)]
+    pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
```

Comment:
We should make this only callable by the PL0 PAUSER. [Example](https://github.com/chipsalliance/caliptra-sw/blob/main/runtime/src/certify_key_extended.rs#L42-L48)

Reply:
Done!